packages/cursorless-engine/src/actions/Wrap.ts

```diff
@@ -50,67 +45,45 @@ export default class Wrap {
-                start,
-                RangeExpansionBehavior.openClosed,
-              ),
-              getSelectionInfo(
-                document,
-                end,
-                RangeExpansionBehavior.closedOpen,
-              ),
-            ];
-          },
-        );
-
-        const cursorSelectionInfos = editor.selections.map((selection) =>
-          getSelectionInfo(
-            document,
-            selection,
-            RangeExpansionBehavior.closedClosed,
-          ),
-        );
+        const boundariesStartSelections = boundaries.map(({ start }) => start);
+        const boundariesEndSelections = boundaries.map(({ end }) => end);
 
-        const sourceMarkSelectionInfos = targets.map((target) =>
-          getSelectionInfo(
-            document,
-            target.contentSelection,
-            RangeExpansionBehavior.closedClosed,
-          ),
-        );
+        const editableEditor = ide().getEditableTextEditor(editor);
 
-        const thatMarkSelectionInfos = targets.map((target) =>
-          getSelectionInfo(
-            document,
-            target.contentSelection,
-            RangeExpansionBehavior.openOpen,
-          ),
+        const contentSelections = targets.map(
+          (target) => target.contentSelection,
         );
 
-        const editableEditor = ide().getEditableTextEditor(editor);
-
-        const [
-          delimiterSelections,
-          cursorSelections,
-          sourceMarkSelections,
-          thatMarkSelections,
-        ] = await performEditsAndUpdateFullSelectionInfos(
-          this.rangeUpdater,
-          editableEditor,
+        const {
+          boundariesStartSelections: delimiterStartSelections,
+          boundariesEndSelections: delimiterEndSelections,
+          sourceSelections: sourceMarkSelections,
+          thatSelections: thatMarkSelections,
+        } = await performEditsAndUpdateSelections({
+          rangeUpdater: this.rangeUpdater,
+          editor: editableEditor,
           edits,
-          [
-            delimiterSelectionInfos,
-            cursorSelectionInfos,
-            sourceMarkSelectionInfos,
-            thatMarkSelectionInfos,
-          ],
-        );
+          selections: {
+            boundariesStartSelections: {
+              selections: boundariesStartSelections,
+              behavior: RangeExpansionBehavior.openClosed,
+            },
+            boundariesEndSelections: {
+              selections: boundariesEndSelections,
+              behavior: RangeExpansionBehavior.closedOpen,
+            },
+            sourceSelections: contentSelections,
+            thatSelections: {
+              selections: contentSelections,
+              behavior: RangeExpansionBehavior.openOpen,
+            },
+          },
+        });
 
-        await editableEditor.setSelections(cursorSelections);
+        const delimiterSelections = [
+          ...delimiterStartSelections,
+          ...delimiterEndSelections,
+        ];
```

Comment:
ah interesting. our interface doesn't allow for a single list to have multiple range behaviours. I think your solution here is fine. Not worth making our interface more complex

Reply:
This is the only example of this so I think it's fine. We don't want increased complexity.